resources/serverless/templates/secrets.yaml

```diff
@@ -1,28 +1,35 @@
-{{- $username := include "tplValue" ( dict "value" .Values.secret.registryUserName "context" . ) -}}
-{{- $password := include "tplValue" ( dict "value" .Values.secret.registryPassword "context" . ) -}}
-{{- $externalRegistry := include "tplValue" ( dict "value" .Values.secret.externalRegistry "context" . ) -}}
+{{- $username := include "tplValue" ( dict "value" .Values.dockerRegistry.username "context" . ) -}}
+{{- $password := include "tplValue" ( dict "value" .Values.dockerRegistry.password "context" . ) -}}
+{{- $internalServerAddress := include "tplValue" ( dict "value" .Values.dockerRegistry.internalServerAddress "context" . ) -}}
+{{- $serverAddress := include "tplValue" ( dict "value" .Values.dockerRegistry.serverAddress "context" . ) -}}
 {{- $encodedUsernamePassword := printf "%s:%s" $username $password | b64enc }}
 apiVersion: v1
 kind: Secret
-type: kubernetes.io/basic-auth
+type: kubernetes.io/dockerconfigjson
+data:
+{{- if .Values.dockerRegistry.enableInternal }}
+  .dockerconfigjson: "{{- (printf "{\"auths\": {\"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}, \"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}}}" $internalServerAddress $encodedUsernamePassword $username $password $serverAddress $encodedUsernamePassword $username $password) | b64enc }}"
+{{- else }}
+  .dockerconfigjson: "{{- (printf "{\"auths\": {\"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}}}" $serverAddress $encodedUsernamePassword $username $password) | b64enc }}"
+{{- end }}
 metadata:
-  name: {{ template "fullname" . }}-registry-credentials
+  name: {{ template "fullname" . }}-registry-docker-config
```

Comment:
I don't remember right now, but will controller also cleanup secrets propagated to namespaces?

Reply:
Changed to older name.